netjsonconfig/backends/openwrt/schema.py

```diff
@@ -8,6 +8,22 @@
 
 default_radio_driver = "mac80211"
 
+# pattern to match ipv4 and ipv6 in CIDR notation
+src_pattern = (
+    "^(s*((([0-9A-Fa-f]{1,4}:){7}([0-9A-Fa-f]{1,4}|:))|(([0-9A-Fa-f]{1,4}:){6}(:[0-9A-Fa-f]{1,4}|"
+    "((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3})|:))|"
+    "(([0-9A-Fa-f]{1,4}:){5}(((:[0-9A-Fa-f]{1,4}){1,2})|:((25[0-5]|2[0-4]d|1dd|[1-9]?d)"
+    "(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3})|:))|(([0-9A-Fa-f]{1,4}:){4}(((:[0-9A-Fa-f]{1,4}){1,3})|"
+    "((:[0-9A-Fa-f]{1,4})?:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|"
+    "(([0-9A-Fa-f]{1,4}:){3}(((:[0-9A-Fa-f]{1,4}){1,4})|((:[0-9A-Fa-f]{1,4}){0,2}:"
+    "((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){2}"
+    "(((:[0-9A-Fa-f]{1,4}){1,5})|((:[0-9A-Fa-f]{1,4}){0,3}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)"
+    "(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){1}(((:[0-9A-Fa-f]{1,4}){1,6})|"
+    "((:[0-9A-Fa-f]{1,4}){0,4}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|"
+    "(:(((:[0-9A-Fa-f]{1,4}){1,7})|((:[0-9A-Fa-f]{1,4}){0,5}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)"
+    "(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:)))(%.+)?s*(\\/(12[0-8]|1[0-1][0-9]|[1-9][0-9]|[0-9])))|"
+    "(((^|\\.)((25[0-5])|(2[0-4]\\d)|(1\\d\\d)|([1-9]?\\d))){4}\\/(?:\\d|[12]\\d|3[01]))$"
+)
```

Comment:
This is huge!
Let's simplify, we don't need to explicitly support ipv4 or ipv6, because it is possible to express a single IP using the CIDR notation, eg:

`192.168.1.2/32` is equivalent to `192.168.1.2`

Please shorten this as much as you can, because this is shown to the user in openwisp-controller.

![Screenshot from 2020-07-11 12-58-30](https://user-images.githubusercontent.com/841044/87230662-1f303900-c377-11ea-9fde-bc8eb08d9758.png)

Reply:
Instead of these self defined pattern, I could just use the jsonschema ipv4 and ipv6 format validation. But the problem is that these formats doesn't support cidr notation. That's 192.168.1.2/32 will be invalid while 192.168.1.2 will be valid. And this will go against the description which states that `src` should be in cidr notation. 

Should I use these formats or try to shorten the one above which ensures the `src and dest` are in valid cidr notations